packages/frontend/src/app/main/repo-list/repo-list.component.ts

```diff
@@ -12,44 +12,22 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {Component, OnInit, Input, ViewChild} from '@angular/core';
-import {PageEvent, MatPaginator} from '@angular/material/paginator';
+import {Component, Input, ViewChild} from '@angular/core';
+import {MatPaginator} from '@angular/material/paginator';
 import {Repository} from 'src/app/services/search/interfaces';
+import {PaginatedListComponent} from 'src/app/paginated-list/paginated-list.component';
 
 @Component({
   selector: 'app-repo-list',
   templateUrl: './repo-list.component.html',
   styleUrls: ['./repo-list.component.css'],
 })
-export class RepoListComponent implements OnInit {
+export class RepoListComponent extends PaginatedListComponent<Repository> {
   @ViewChild('paginator') paginator: MatPaginator;
 
-  _repositories: Repository[] = [];
   @Input() set repositories(value: Repository[]) {
-    this._repositories = value;
+    this._elements = value;
     this.updatePage();
     this.paginator?.firstPage();
   }
-
-  renderedRepositories: Repository[] = [];
-  pageIndex = 0;
```

Comment:
This unmerged PR adds support for server side pagination. on the /api/repo endpoint there is a limit, startsaftername, and endsbeforename parameter, so if you want to go to the next page pass in startsaftername as an input with the last name on the current page. 

https://github.com/GoogleCloudPlatform/flaky-service/pull/54

Reply:
Got it 👍
Knowing that, my strategy was to fetch a max number of data (ie: 1000) and when/if the user reaches the last page (or a page close to the last one), the client will request more data based on the name of the last repo.
**In other words, startsaftername and endsbeforename will be perfect !** 👌